File: presentation/src/main/java/pl/mobly/simplylauncher/common/components/modules/DataModule.java
package pl.mobly.simplylauncher.common.components.modules;

import android.app.Application;
import android.content.Context;
import android.content.pm.PackageManager;

import dagger.Module;
import dagger.Provides;
import pl.mobly.data.PackageManagerResource;
import pl.mobly.data.SystemApiImpl;

@Module
public class DataModule {
    private Application application;

    public DataModule(Application application) {
        this.application = application;
    }

    @Provides
    PackageManagerResource providePackageManagerResource(PackageManager packageManager) {
        return new PackageManagerResource(packageManager);
    }

    @Provides
    SystemApiImpl providesSystemApi(PackageManagerResource packageManagerResource) {
    return new SystemApiImpl(packageManagerResource);
    }
}

File: presentation/src/main/java/pl/mobly/simplylauncher/common/components/modules/DataModule.kt
package pl.mobly.simplylauncher.common.components.modules

import android.app.Application
import android.content.pm.PackageManager
import dagger.Module
import dagger.Provides
import pl.mobly.data.PackageManagerResource
import pl.mobly.data.SystemApiImpl
import javax.inject.Singleton


@Module class DataModule(private val application: Application) {

    @Provides
    @Singleton
    fun providePackageManagerResource(packageManager: PackageManager) : PackageManagerResource {
        return PackageManagerResource(packageManager)
    }

    @Provides
    @Singleton
    fun provideSystemApi(packageManagerResource: PackageManagerResource) : SystemApiImpl {
        return SystemApiImpl(packageManagerResource)
    }
}